ffi/src/owned.rs

```diff
@@ -161,18 +146,36 @@ impl<'itm, T> IntoFfiVec for &'itm T
 where
     &'itm T: IntoFfi,
 {
-    type Target = LocalSlice<<Self as IntoFfi>::Target>;
+    type Target = SliceRef<'itm, <Self as IntoFfi>::Target>;
+    type Store = (
+        Vec<<&'itm T as IntoFfi>::Target>,
+        Vec<MaybeUninit<<&'itm T as IntoFfi>::Store>>,
+    );
+
+    fn into_ffi(source: Vec<Self>, store: &mut MaybeUninit<Self::Store>) -> Self::Target {
+        let mut store = store.write((
+            Vec::new(),
+            core::iter::repeat_with(MaybeUninit::uninit)
+                .take(source.len())
+                .collect(),
+        ));
 
-    fn into_ffi(source: Vec<Self>) -> Self::Target {
-        source.into_iter().map(IntoFfi::into_ffi).collect()
+        store.0 = source
+            .into_iter()
+            .enumerate()
+            .map(|(i, item)| IntoFfi::into_ffi(item, &mut store.1[i]))
+            .collect();
+
+        SliceRef::from_slice(&store.0)
```

Comment:
This mean that `Target` will be valid only for lifetime of `Store`, right?

So for example user can't safely return `Target` from function call if previously allocate `Store` on stack.

Reply:
exactly, you're on it. Only types where `Target = ()` are ever allowed to be returned